sound/soc/sof/xtensa/core.c

```diff
@@ -132,6 +132,17 @@ static void xtensa_stack(struct snd_sof_dev *sdev, const char *level, void *oops
 				   buf, sizeof(buf), false);
 		dev_printk(level, sdev->dev, "0x%08x: %s\n", stack_ptr + i * 4, buf);
 	}
+
+	if (!xoops->plat_hdr.numaregs)
+		return;
+
+	dev_printk(level, sdev->dev, "AR registers:\n");
+	/* the number of ar registers is a multiple of 4 */
+	for (i = 0; i < xoops->plat_hdr.numaregs; i += 4) {
```

Comment:
Sorry for the late catch. but what if the panic is corrupted. I guess we should have a upper limit for "numaregs", so this cannot go outside the maximum dump size and start dumping out random kernel memory out.

Reply:
Thanks !  updated. According to Xtensa spec, dsp has AR0 to AR15, so I add max AR check